std/expand_fixed.asm

```diff
@@ -0,0 +1,77 @@
+use std::result::Result;
+use std::check::panic;
+
+// A term in an array expression
+enum ArrayTerm {
+    Repeat(int[]),
+    Once(int[])
+}
+
+// returns the total size of the repeated array in this array expression
+let solve: ArrayTerm[], int -> Result<int, string> = |terms, degree| {
```

Comment:
`solve` sounds a bit too generic for me. What about something like `compute_length`?

Reply:
ah, `compute_length_of_repeated_part`